plugin/pkg/utils/cloudprofile_test.go

```diff
@@ -0,0 +1,310 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package utils_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	"github.com/gardener/gardener/pkg/apis/core"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardencoreinformers "github.com/gardener/gardener/pkg/client/core/informers/externalversions"
+	gardencorev1beta1listers "github.com/gardener/gardener/pkg/client/core/listers/core/v1beta1"
+	"github.com/gardener/gardener/pkg/features"
+	"github.com/gardener/gardener/pkg/utils/test"
+	admissionutils "github.com/gardener/gardener/plugin/pkg/utils"
+)
+
+var _ = Describe("CloudProfile", func() {
+	var (
+		coreInformerFactory          gardencoreinformers.SharedInformerFactory
+		cloudProfileLister           gardencorev1beta1listers.CloudProfileLister
+		namespacedCloudProfileLister gardencorev1beta1listers.NamespacedCloudProfileLister
+
+		namespaceName              = "foo"
+		cloudProfileName           = "profile-1"
+		namespacedCloudProfileName = "n-profile-1"
+
+		cloudProfile           *gardencorev1beta1.CloudProfile
+		namespacedCloudProfile *gardencorev1beta1.NamespacedCloudProfile
+
+		shoot *core.Shoot
+	)
+
+	BeforeEach(func() {
+		coreInformerFactory = gardencoreinformers.NewSharedInformerFactory(nil, 0)
+		cloudProfileLister = coreInformerFactory.Core().V1beta1().CloudProfiles().Lister()
+		namespacedCloudProfileLister = coreInformerFactory.Core().V1beta1().NamespacedCloudProfiles().Lister()
+
+		cloudProfile = &gardencorev1beta1.CloudProfile{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: cloudProfileName,
+			},
+		}
+
+		namespacedCloudProfile = &gardencorev1beta1.NamespacedCloudProfile{
+			Spec: gardencorev1beta1.NamespacedCloudProfileSpec{
+				Parent: gardencorev1beta1.CloudProfileReference{
+					Kind: "CloudProfile",
+					Name: cloudProfileName,
+				},
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      namespacedCloudProfileName,
+				Namespace: namespaceName,
+			},
+		}
+
+		shoot = &core.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespaceName,
+			},
+		}
+	})
+	Describe("#GetCloudProfile", func() {
```

Comment:
Shouldn't this and the following `Describe`s not be nested in `Describe("CloudProfile"`?

Reply:
It's just another `Describe` layer to have each function tested in a separate block. Do you still want me to pull one block out, though?

Reply:
My bad, the structure is fine 🙂